### DISCUSSION TITLE
Working on amd64 ubuntu

### DISCUSSION THREAD

**Discussion by swt006** [zmp/robocar110_ros#8] [Q&A (問答)] [answered]
Is it able to install and execute the souce code on an amd64 ubuntu system?
I have trouble installing ROS Node due to the difference between arm64 and amd64.
Could you help me with that?

**Comment by z-andrei-pak** [comment, marked as answer]
@swt006 Sorry for the late reply. My watch settings were not configured properly.

> Is it able to install and execute the souce code on an amd64 ubuntu system?

Yes. (Thanks for the question, I'll update the README.)

> Could you help me with that?

Please, describe your problem in more details.

**Reply by swt006** [reply]
Thank you for your reply.
Right now I am trying to run the source code on ZMP Robocar 1/10, last generation without gpu and is amd system.
I have upgraded the ubuntu system to 18.04, and installed ROS system. But I face two problems.

1.When running command to install ROS node
sh rc110_robot_*.run
[error1.md](https://github.com/zmp/robocar110_ros/files/9280396/error1.md)

2.When running commands to start the source code
make
[error2.md](https://github.com/zmp/robocar110_ros/files/9280395/error2.md)

Look forward to reply.

**Reply by z-andrei-pak** [reply]
I see...

`*.run` files and `make` targets are only for running on RoboCar 1/10 X robot.

Previous models are not supported, because underlying driver was not updated for them.

AMD64 is only supported for remote PC: https://github.com/zmp/robocar110_ros/blob/master/docs/RemoteConnection.md

**Reply by z-andrei-pak** [reply]
I suggest to contact our sales about this question: https://www.zmp.co.jp/en/contact/inquiry

**Reply by z-andrei-pak** [reply]
Sorry for the misleading title of the repo.
Initially it was considered to support both amd64 and arm64 robots, but until now there was no demand to do so.

**Reply by swt006** [reply]
Thank you for your clarification. 
Is there any way to update the driver?
Also I would like to learn how to support amd64 robots.

**Reply by z-andrei-pak** [reply]
Hm... It's my personal opinion, not official company reply. Because it's quite a lot of work to update the C driver, either we need more people who ask about it, or just one person who is willing to pay for the development. For the official reply, please, do the inquiry as described above.

**Reply by swt006** [reply]
Okay, I will take it into consideration. Anyway, thank you for your reply.
